internal/annotators/http/parser.go

```diff
@@ -0,0 +1,115 @@
+	fmt.Printf("Signature-Input list: %v\n", signatureInputList)
+
+	signatureInputFields := make(map[string][]string)
+	for i := 0; i < len(signatureInputList); i += 1 {
+		field := signatureInputList[i]
+		//remove double quotes from the field to access it directly in the header map
+		signatureInputList[i] = field[1 : len(field)-1]
+		signatureInputFields[signatureInputList[i]] = []string{}
+	}
+
+	for headerField := range signatureInputFields {
+		// Check if it is an actual header field, or a specialty component
+		if headerField[0:1] == "@" {
+			switch headerField {
+			case "@method":
+				signatureInputFields[headerField] = []string{r.Method}
+			case "@authority":
+				signatureInputFields[headerField] = []string{r.Host}
+			case "@scheme":
+				protool := r.Proto
+				scheme := strings.ToLower(strings.Split(protool, "/")[0])
+				signatureInputFields[headerField] = []string{scheme}
+			case "@request-target":
+				signatureInputFields[headerField] = []string{r.RequestURI}
+			case "@path":
+				signatureInputFields[headerField] = []string{r.URL.Path}
+			case "@query":
+				var query string = "?"
+				query += r.URL.RawQuery
+				signatureInputFields[headerField] = []string{query}
+			case "@query-params":
+				queryParamsRawMap := r.URL.Query()
+				var queryParams []string
+				for key, value := range queryParamsRawMap {
+					b := new(bytes.Buffer)
+					fmt.Fprintf(b, ";name=\"%s\": %s", key, value[0])
+					queryParams = append(queryParams, b.String())
+				}
+
+				signatureInputFields[headerField] = queryParams
+			default:
+				fmt.Println("Unhandled Specialty Component")
+			}
+
+			// handle eader fields
+		} else {
+			fieldValues := r.Header.Values(headerField)
+
+			if len(fieldValues) == 1 {
+				value := RemoveExtraSpaces(r.Header.Get(headerField))
+				signatureInputFields[headerField] = []string{value}
+
+			} else {
+
+				value := ""
+				for i := 0; i < len(fieldValues); i++ {
+					value += fieldValues[i]
+					if i != (len(fieldValues) - 1) {
+						value += ", "
+					}
+				}
+				value = RemoveExtraSpaces(value)
+				signatureInputFields[headerField] = []string{value}
+			}
+		}
+	}
+
+	fmt.Printf("signatureInputFields: %v\n", signatureInputFields)
+
+	// Construct final output string
+	deserializedSignatureInput := ""
+	for i := 0; i < len(signatureInputList); i += 1 {
+		field := signatureInputList[i]
+		value := signatureInputFields[field]
+		if len(value) == 1 {
+			deserializedSignatureInput += ("\"" + field + "\": " + value[0] + "\n")
+		} else {
+			for _, v := range value {
+				deserializedSignatureInput += ("\"" + field + "\"" + v + "\n")
+			}
+		}
+	}
+	//remove signature name
+	index := strings.Index(signatureInput, "=")
+	signatureInput = signatureInput[index+1:]
+
+	// check if new line needs to be added at the end
+	deserializedSignatureInput += ("\"@signature-params\": " + signatureInput + "\n")
+	fmt.Printf("FINAL OUTPUT %v\n", deserializedSignatureInput)
+
+	return deserializedSignatureInput
```

Comment:
This value is not used by the caller currently. I presume that is TBD and will replace how the signature is currently constructed.

Reply:
The returned string here is equal to the `seed` in the regular PKI annotator. We are still looking into the best way to parse the `signature` string.

Reply:
addressed in 6a1ab4968bd3227d6660855a714724cd3d0633a5